### ISSUE TITLE
Packer shuts down Virtualbox machine right after starting to upload Puppet modules

### ISSUE THREAD

**Issue by seanhagen** [hashicorp/packer#863]
Packer version: 0.5.1
VirtualBox version: 4.1.12_Ubuntur77245

Log gist: https://gist.github.com/seanhagen/8701082
Config gist: https://gist.github.com/seanhagen/8701111

I'm trying to build a custom Vagrant box image, and I'm running into an issue I can't seem to figure out. Towards the end of the set up run, I see these lines in the log:

```
2014/01/29 17:37:14 /home/sean/bin/packer-builder-virtualbox-iso: 2014/01/29 17:37:14 Beginning file upload...
2014/01/29 17:37:14 /home/sean/bin/packer-builder-virtualbox-iso: 2014/01/29 17:37:14 SCP: starting directory upload: modules
2014/01/29 17:37:14 /home/sean/bin/packer-builder-virtualbox-iso: 2014/01/29 17:37:14 Executing VBoxManage: []string{"controlvm", "packer-virtualbox-iso", "poweroff"}
2014/01/29 17:37:18 /home/sean/bin/packer-builder-virtualbox-iso: 2014/01/29 17:37:18 stdout:
2014/01/29 17:37:18 /home/sean/bin/packer-builder-virtualbox-iso: 2014/01/29 17:37:18 stderr: 0%...10%...20%...30%...40%...50%...60%...70%...80%...90%...100%
2014/01/29 17:37:18 /home/sean/bin/packer-builder-virtualbox-iso: 2014/01/29 17:37:18 Executing VBoxManage: []string{"storageattach", "packer-virtualbox-iso", "--storagectl", "IDE Controller", "--port", "0", "--device", "1", "--medium", "none"}
```

From what I see in the log, it looks like it's trying to upload the Puppet modules AND shut down the VM at the same time. Once it's done that, it tries to delete the VM:

```
==> virtualbox-iso: Unregistering and deleting virtual machine...
2014/01/29 17:37:18 ui: ==> virtualbox-iso: Unregistering and deleting virtual machine...
2014/01/29 17:37:18 /home/sean/bin/packer-builder-virtualbox-iso: 2014/01/29 17:37:18 Executing VBoxManage: []string{"unregistervm", "packer-virtualbox-iso", "--delete"}
2014/01/29 17:37:18 /home/sean/bin/packer-builder-virtualbox-iso: 2014/01/29 17:37:18 stdout:
2014/01/29 17:37:18 /home/sean/bin/packer-builder-virtualbox-iso: 2014/01/29 17:37:18 stderr: VBoxManage: error: Cannot unregister the machine 'packer-virtualbox-iso' while it is locked
2014/01/29 17:37:18 /home/sean/bin/packer-builder-virtualbox-iso: VBoxManage: error: Details: code VBOX_E_INVALID_OBJECT_STATE (0x80bb0007), component Machine, interface IMachine, callee nsISupports
2014/01/29 17:37:18 /home/sean/bin/packer-builder-virtualbox-iso: Context: "Unregister(fDelete ? (CleanupMode_T)CleanupMode_DetachAllReturnHardDisksOnly : (CleanupMode_T)CleanupMode_DetachAllReturnNone, ComSafeArrayAsOutParam(aMedia))" at line 160 of file VBoxManageMisc.cpp
==> virtualbox-iso: Error deleting virtual machine: VBoxManage error: VBoxManage: error: Cannot unregister the machine 'packer-virtualbox-iso' while it is locked
==> virtualbox-iso: VBoxManage: error: Details: code VBOX_E_INVALID_OBJECT_STATE (0x80bb0007), component Machine, interface IMachine, callee nsISupports
==> virtualbox-iso: Context: "Unregister(fDelete ? (CleanupMode_T)CleanupMode_DetachAllReturnHardDisksOnly : (CleanupMode_T)CleanupMode_DetachAllReturnNone, ComSafeArrayAsOutParam(aMedia))" at line 160 of file VBoxManageMisc.cpp
==> virtualbox-iso: Deleting output directory...
2014/01/29 17:37:18 ui error: ==> virtualbox-iso: Error deleting virtual machine: VBoxManage error: VBoxManage: error: Cannot unregister the machine 'packer-virtualbox-iso' while it is locked
==> virtualbox-iso: VBoxManage: error: Details: code VBOX_E_INVALID_OBJECT_STATE (0x80bb0007), component Machine, interface IMachine, callee nsISupports
==> virtualbox-iso: Context: "Unregister(fDelete ? (CleanupMode_T)CleanupMode_DetachAllReturnHardDisksOnly : (CleanupMode_T)CleanupMode_DetachAllReturnNone, ComSafeArrayAsOutParam(aMedia))" at line 160 of file VBoxManageMisc.cpp
2014/01/29 17:37:18 ui: ==> virtualbox-iso: Deleting output directory...
Build 'virtualbox-iso' errored: Error uploading modules: lstat /Users: no such file or directory
2014/01/29 17:37:18 ui error: Build 'virtualbox-iso' errored: Error uploading modules: lstat /Users: no such file or directory
2014/01/29 17:37:18 /home/sean/bin/packer-command-build: 2014/01/29 17:37:18 Builds completed. Waiting on interrupt barrier...
2014/01/29 17:37:18 machine readable: error-count []string{"1"}

==> Some builds didn't complete successfully and had errors:
2014/01/29 17:37:18 ui error: 
==> Some builds didn't complete successfully and had errors:
2014/01/29 17:37:18 machine readable: virtualbox-iso,error []string{"Error uploading modules: lstat /Users: no such file or directory"}
--> virtualbox-iso: Error uploading modules: lstat /Users: no such file or directory
2014/01/29 17:37:18 ui error: --> virtualbox-iso: Error uploading modules: lstat /Users: no such file or directory
```

Do I need to use more specific configuration settings ( http://www.packer.io/docs/provisioners/puppet-masterless.html ) in the Puppet provisioner section for Packer, or am I missing something else? I see that the last error that Packer outputs is:

```
Error uploading modules: lstat /Users: no such file or directory
```

I'm wondering if I need to use the file provisioner to upload the files first, and use the puppet provisioner settings to point puppet there.

Any help much appreciated.

**Comment by lmayorga1980** [comment]
:+1: Seems like the issue persist when a puppet module dependency contains symlinks

**Comment by rickard-von-essen** [comment]
There wasn't any issue with packer.

**Comment by lmayorga1980** [comment]
You are right. But from the user perspective the following message...

`Error uploading modules: lstat /Users: no such file or directory`

Does not help the user to go a find the `symlink` inside the module dependencies and clean them properly.  Maybe a feature during `packer validate` step where

1. `puppet-masterless` is being used.
2. the puppet modules directory contains a symlink.
3. print some `warning`

**Comment by rickard-von-essen** [comment]
Could you open a new issue describing your feature request.
